app/models/checkin.rb

```diff
@@ -35,31 +38,13 @@ def self.limit_returned_checkins(args)
     if args[:action] == 'index' && args[:multiple_devices]
       all
     elsif args[:action] == 'index' && !args[:multiple_devices]
-      paginate(page: args[:page], per_page: args[:per_page])
+      per_page = all.length > 1 ? args[:per_page] : 1
+      paginate(page: args[:page], per_page: per_page)
```

Comment:
Have to do this as otherwise if privilege=last_only, but they ask for 30 checkins per_page, the limit(1) applied by the privilege is overriden by the limit(per_page).